File: common/src/main/java/com/leo/commonutil/http/Respond.java
package com.leo.commonutil.http;

import android.os.Parcel;
import android.os.Parcelable;

public class Respond implements Parcelable {
    private int code;
    private String body;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.code);
        dest.writeString(this.body);
    }

    public Respond() {
    }

    protected Respond(Parcel in) {
        this.code = in.readInt();
        this.body = in.readString();
    }

    public static final Parcelable.Creator<Respond> CREATOR = new Parcelable.Creator<Respond>() {
        @Override
        public Respond createFromParcel(Parcel source) {
            return new Respond(source);
        }

        @Override
        public Respond[] newArray(int size) {
            return new Respond[size];
        }
    };
}

File: common/src/main/java/com/leo/commonutil/http/Respond.kt
package com.leo.commonutil.http

import android.os.Parcel
import android.os.Parcelable

data class Respond(var code: Int, var body: String?) : Parcelable {

    constructor() : this(0, "") {

    }

    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(code)
        parcel.writeString(body)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Respond> {
        override fun createFromParcel(parcel: Parcel): Respond {
            return Respond(parcel)
        }

        override fun newArray(size: Int): Array<Respond?> {
            return arrayOfNulls(size)
        }
    }

}